### PR TITLE
s3config replace login for xs3 auth UX flow

### PR DESCRIPTION
**Pull request by seanses**
Fix CLI-164.

Run `git xet s3config [--host ..]` to pick up login information from ~/.xetconfig to create a env file for AWS CLI. Users will source this env file to pick up `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_ENDPOINT_URL`. After that users can directly run `aws s3 ...` to operate with XS3.

- **If no access_key or secret_key found in the .xetconfig corresponding profile, print**

```
No XetHub authentication found.

Create a new personal access token at https://xethub.com/user/settings/pat
and run the displayed `git xet login` command to authenticate.

Re-try your s3config command after logging in.
```

- **If no xs3 server address found in the .xetconfig corresponding profile ( this can happen if either**
          **1. XS3 service is not available for this XetHub deployment;**
          **2. People who logged in previously but didn't have the xs3 server endpoint returned from Xetea and stored in the .xetconfig),** 
**print**

```
No XetHub S3 service was found for this deployment.

If you believe this to be an error, create a new token at https://xethub.com/user/settings/pat
and run the displayed `git xet login` command to authenticate. Then re-run your original command.

Still getting this error? Reach out to contact@xethub.com or your administrator for support.
```

- **If everything finishes correctly, print**

```
XetHub S3 service profile configuration complete.

Source the environment file to initialize:
. "/Users/di/.xet/s3env"

Confirm success by listing the XetHub repositories you can access:
aws s3 ls
```

Depends on https://github.com/xetdata/xethub/pull/5875

### CODE REVIEW

**Comment by seanses** [comment]
According to [this](https://linuxcommand.org/lc3_adv_source.php#:~:text=While%20bash%2C%20ksh%2C%20and%20zsh%20all%20have%20the%20source%20builtin%2C%20dash%20and%20all%20other%20strictly%20POSIX%20compatible%20shells%20support%20only%20the%20single%20dot%20(.).)

> While bash, ksh, and zsh all have the source builtin, dash and all other strictly POSIX compatible shells support only the single dot (.).

File: rust/gitxetcore/src/command/s3config.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    let default_path_str = format!("$HOME/{DEFAULT_XET_HOME}/{S3_CONFIG_FILE_NAME}");
+    #[cfg(windows)]
+    let default_path_str =
+        format!("%HOMEDRIVE%%HOMEPATH%\\{DEFAULT_XET_HOME}\\{S3_CONFIG_FILE_NAME}");
+    let path_str = s3config_path.to_str().unwrap_or(&default_path_str);
+    #[cfg(unix)]
+    eprintln!(
+        r#"Profile for the XetHub S3 Service is configured now. Great!
+
+1. To get started source the corresponding env file. This is usually done by
```
Comment (hoytak) [inline rust/gitxetcore/src/command/s3config.rs:70]: Nit:  To get started, source the corresponding env file.

File: rust/gitxetcore/src/command/s3config.rs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+running one of the following (note the leading DOT):
+. "{}"                            # For sh/bash/zsh/ash/dash/pdksh
+source "{}"                       # For fish
```
Comment (hoytak) [inline rust/gitxetcore/src/command/s3config.rs:73]: Not source we want to do it, but "source" works everywhere, `.` is an alias in all posix shells.  Might be simpler to just go with `source`

File: rust/gitxetcore/src/command/s3config.rs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    let default_path_str =
+        format!("%HOMEDRIVE%%HOMEPATH%\\{DEFAULT_XET_HOME}\\{S3_CONFIG_FILE_NAME}");
+    let path_str = s3config_path.to_str().unwrap_or(&default_path_str);
+    #[cfg(unix)]
+    eprintln!(
+        r#"Profile for the XetHub S3 Service is configured now. Great!
+
+1. To get started source the corresponding env file. This is usually done by
+running one of the following (note the leading DOT):
+. "{}"                            # For sh/bash/zsh/ash/dash/pdksh
```
Comment (hoytak) [inline rust/gitxetcore/src/command/s3config.rs:72]: Can do `{path_str:<32}` to put path_str left-aligned and padded out with spaces to 32 character width if less. (noting the alignment above.

File: rust/gitxetcore/src/command/s3config.rs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        "#,
+        path_str, path_str,
+    );
+    #[cfg(windows)]
+    eprintln!(
+        r#"1. Profile for the XetHub S3 Service is configured now. Great!
+
+To get started source the corresponding env file. This can be done by running
+the following:
+"{}"                              # For cmd
```
Comment (hoytak) [inline rust/gitxetcore/src/command/s3config.rs:83]: Missing `source` ?

Comment (seanses) [reply rust/gitxetcore/src/command/s3config.rs:83]: Actually no. On Windows this will be a `bat` file which should be directly executed.

**Comment by erinys** [comment]
> According to [this](https://linuxcommand.org/lc3_adv_source.php#:~:text=While%20bash%2C%20ksh%2C%20and%20zsh%20all%20have%20the%20source%20builtin%2C%20dash%20and%20all%20other%20strictly%20POSIX%20compatible%20shells%20support%20only%20the%20single%20dot%20(.).)
> 
> > While bash, ksh, and zsh all have the source builtin, dash and all other strictly POSIX compatible shells support only the single dot (.).
> 
> If we want to unify the below section I opt for "." instead of "source". What do you think? @erinys @hoytak
> 
> ```
> . "/Users/di/.xet/s3env"                            # For sh/bash/zsh/ash/dash/pdksh
> source "/Users/di/.xet/s3env"                       # For fish
> ```

I'm ok with just using dot if it also works for fish (which I've never even heard of 😅).

File: rust/gitxetcore/src/command/s3config.rs [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+fn print_help_message(s3config_path: &Path) {
+    #[cfg(unix)]
+    let default_path_str = format!("$HOME/{DEFAULT_XET_HOME}/{S3_CONFIG_FILE_NAME}");
+    #[cfg(windows)]
+    let default_path_str =
+        format!("%HOMEDRIVE%%HOMEPATH%\\{DEFAULT_XET_HOME}\\{S3_CONFIG_FILE_NAME}");
+    let path_str = s3config_path.to_str().unwrap_or(&default_path_str);
+    #[cfg(unix)]
+    eprintln!(
+        r#"Profile for the XetHub S3 Service is configured now. Great!
```
Comment (erinys) [inline rust/gitxetcore/src/command/s3config.rs:68]: Maybe simplify the wording a bit:

```
XetHub S3 service profile configuration complete.

Source the s3 environment file to initialize:
. <path/to/file>

Confirm success by listing the XetHub repositories you can access:
aws s3 ls
```

File: rust/gitxetcore/src/command/s3config.rs [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+fn print_auth_error_message(host: &str) {
+    eprintln!(
+        r#"No authentication information found.
+
+Please go to https://{}/user/settings/pat to create a token and 
+run `git xet login -u <user_name> -e <email> -p <PAT>{}` to login. 
+
+After that, re-run this command."#,
```
Comment (erinys) [inline rust/gitxetcore/src/command/s3config.rs:100]: Minor wording suggestion:

```
No XetHub authentication found.

Create a new personal access token at https://xethub.com/user/settings/pat
and run the displayed `git xet login` command to authenticate. 

Re-try your s3config command after logging in. 
```

File: rust/gitxetcore/src/command/s3config.rs [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+            format!(" --host {host}")
+        }
+    );
+}
+
+fn print_service_unavailable_message(host: &str) {
+    eprintln!(
+        r#"A XetHub S3 service is not available for this XetHub deployment.
+
+If you believe this is a mistake, try re-run `git xet login` with your credentials
```
Comment (erinys) [inline rust/gitxetcore/src/command/s3config.rs:114]: Wording suggestion:

```
No XetHub s3 service was found for this deployment.

If you believe this to be an error, create a new token at https://xethub.com/user/settings/pat
and run the displayed `git xet login` command to authenticate.
Then re-run your original command.

Still getting this error? Reach out to contact@xethub.com or your administrator for support.
```

**Review by erinys — CHANGES_REQUESTED** [review]

**Comment by seanses** [comment]
> I'm ok with just using dot if it also works for fish (which I've never even heard of 😅).

The [fish doc](https://fishshell.com/docs/current/cmds/source.html#:~:text=.%20(a%20single%20period)%20is%20an%20alias%20for%20the%20source%20command.%20The%20use%20of%20.%20is%20deprecated%20in%20favour%20of%20source%2C%20and%20.%20will%20be%20removed%20in%20a%20future%20version%20of%20fish.)  says 

> . will be removed in a future version of fish

But let's just use "." here. I believe these advanced shell users should know how to source.

**Review by erinys — APPROVED** [review]
Thanks for the updates!

**Review by hoytak — APPROVED** [review]
